triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -212,7 +212,17 @@ class Runner(
             // This happens for invalid UUIDs which we might get for completions not emitted by the trigger.
             case e: IllegalArgumentException => List()
           }
-        case msg @ TransactionMsg(_) => List(msg)
+        case msg @ TransactionMsg(t) =>
+          try {
+            commandIdMap.get(UUID.fromString(t.commandId)) match {
```

Comment:
The empty case should be handled by the `try catch`. `UUID.fromString` should throw an `IllegalArgumentException` in case of an empty string or otherwise illegal UUID and the `catch` block handles that case.

> I would propose that in that case we should treat this as if no command id has been supplied and do the same for when the lookup fails.

So, clear the `commandId` field instead of forwarding the incoming `msg` unchanged?

Reply:
Ah yeah, the catch makes sense. Clearing `commandId` sounds good and we should we should do thet for completions as well.

Reply:
I've changed it to clear the `commandId` in such cases. The `Converter` will then convert that into a `None : Optional CommandId`.

In case of `MCompletion` we're already dropping those messages with unknown command ids.